lib/topic-alias-send.js

```diff
@@ -0,0 +1,93 @@
+'use strict'
+
+/**
+ * Module dependencies
+ */
+var LruMap = require('collections/lru-map')
+var NumberAllocator = require('number-allocator').NumberAllocator
+
+/**
+ * Topic Alias sending manager
+ * This holds both topic to alias and alias to topic map
+ * @param {Number} [max] - topic alias maximum entries
+ */
+function TopicAliasSend (max) {
+  if (!(this instanceof TopicAliasSend)) {
+    return new TopicAliasSend(max)
+  }
+
+  if (max > 0) {
+    this.aliasToTopic = new LruMap()
+    this.topicToAlias = {}
+    this.numberAllocator = new NumberAllocator(1, max)
+    this.max = max
+    this.length = 0
+  }
+}
+
+/**
+ * Insert or update topic - alias entry.
+ * @param {String} [topic] - topic
+ * @param {Number} [alias] - topic alias
+ * @returns {Boolean} - if success return true otherwise false
+ */
+TopicAliasSend.prototype.put = function (topic, alias) {
+  if (alias === 0 || alias > this.max) {
+    return false
+  }
+  const entry = this.aliasToTopic.get(alias)
+  if (entry) {
+    delete this.topicToAlias[entry.topic]
+  }
+  this.aliasToTopic.set(alias, {'topic': topic, 'alias': alias})
```

Comment:
is this redundant to have the `set` method on `aliasToTopic` and `topicToAlias` as a general JS object?

Reply:
```js
    this.aliasToTopic = new LruMap()
    this.topicToAlias = {}
```

`aliasToTopic` is an instance of LruMap(). `aliasToTopic` requires LRU for overwriting. So it cannot be a simple JS Object.
`topicToAlias` is JS Object. It doesn't require LRU.

Could you elaborate your question?
Do you mean add set method to `topicToAlias` ?
And update the code using `this/topicToAlias.set(topic, alias)` instead of `this.topicToAlias[topic] = alias` ?
I personally think it is redundant.

Reply:
yes that makes sense. Ignore my initial question. New question: why `{topic: topic, alias: alias}`, when the information for the `alias` is the key?

Reply:
This is for 
```js
  return this.aliasToTopic.min().alias
```

See https://github.com/mqttjs/MQTT.js/pull/1301/files/f8178aab47846ba7f2f34e436bc1ff3d5f23de59#diff-ae7ca6f8f9c9488b9789056d3b1e4aee44861bb0f86c3000c8361bbad8c7bb28R90

lru-map only provide least **value** not key. I tried to find getting key-value entry way but it seems that it doesn't exist. https://www.collectionsjs.com/lru-map

That's why I added alias not only as key but also as value.